frontend/src/metabase/search/containers/SearchApp.jsx

```diff
@@ -25,172 +30,90 @@ import {
-                        showTotal
-                        pageSize={PAGE_SIZE}
-                        page={page}
-                        itemsLength={list.length}
-                        total={metadata.total}
-                        onNextPage={handleNextPage}
-                        onPreviousPage={handlePreviousPage}
-                      />
-                    </div>
-                  </Fragment>
-                </SearchMain>
+      <Search.ListLoader query={query} wrapped>
+        {({ list, metadata }) =>
+          list && list.length > 0 ? (
+            <SearchBody>
+              <SearchMain>
+                <SearchResultSection items={list} />
+                <PaginationControls
+                  showTotal
+                  pageSize={PAGE_SIZE}
+                  page={page}
+                  itemsLength={list.length}
+                  total={metadata.total}
+                  onNextPage={handleNextPage}
+                  onPreviousPage={handlePreviousPage}
+                />
+              </SearchMain>
+              {searchFilters?.type || metadata.available_models ? (
                 <SearchControls>
-                  {filters.length > 0 ? (
-                    <Link
-                      className={cx("flex align-center mb3", {
-                        "text-brand": filter == null,
-                        "text-inherit": filter != null,
-                      })}
-                      onClick={() => handleFilterChange(null)}
-                      to={{
-                        pathname: location.pathname,
-                        query: { ...location.query, type: undefined },
-                      }}
-                    >
-                      <Icon name="search" className="mr1" />
-                      <h4>{t`All results`}</h4>
-                    </Link>
-                  ) : null}
-                  {filters.map(f => {
-                    const isActive = filter === f.filter;
-
-                    return (
-                      <Link
-                        key={f.filter}
-                        className={cx("mb3 flex align-center", {
-                          "text-brand": isActive,
-                          "text-medium": !isActive,
-                        })}
-                        onClick={() => handleFilterChange(f)}
-                        to={{
-                          pathname: location.pathname,
-                          query: { ...location.query, type: f.filter },
-                        }}
-                      >
-                        <Icon className="mr1" name={f.icon} size={16} />
-                        <h4>{f.name}</h4>
-                      </Link>
-                    );
-                  })}
+                  <TypeSearchSidebar
+                    availableModels={metadata.available_models.filter(
+                      filter =>
+                        !searchFilters?.type ||
+                        searchFilters.type.includes(filter),
```

Comment:
nit: `type` sound like a single value, I wouldn't expect it to be an array. Can we rename this to `types`?

Also, similar to my other comment above. `searchFilters.type` here means that `searchFilters` will not be null. But based on the condition before (both `searchFilters?.type || metadata.available_models` and `!searchFilters?.type`), it looks like it could be null. 

Can you double-check if this is correct?

Reply:
On the nit - I was thinking that the `SearchFilters` object would reflect the query params that we would be using (i.e. `/search?type=dataset&type=card...`). I think that makes more sense considering that we'll be hydrating and pushing those keys to and from the search URL

Reply:
Answered the other part of this question above